optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/supply/SupplyManager.java

```diff
@@ -22,13 +22,15 @@
 public interface SupplyManager {
 
     /**
-     * Returns the {@link Supply} for a {@link Demand}, preferably an existing one.
-     * If the {@link Supply} doesn't exist yet (as part of the domain model or externalized), it creates and attaches it.
+     * Returns the {@link org.optaplanner.core.impl.domain.variable.supply.Supply} for a {@link Demand}, preferably an existing
+     * one.
+     * If the {@link org.optaplanner.core.impl.domain.variable.supply.Supply} doesn't exist yet (as part of the domain model or
+     * externalized), it creates and attaches it.
      *
      * @param demand never null
-     * @param <S> Subclass of {@link Supply}
+     * @param <Supply> Subclass of {@link org.optaplanner.core.impl.domain.variable.supply.Supply}
      * @return never null
      */
-    <S extends Supply> S demand(Demand<S> demand);
+    <Supply extends org.optaplanner.core.impl.domain.variable.supply.Supply> Supply demand(Demand<Supply> demand);
```

Comment:
First of all, `<Supply_ extends Supply> Supply_` should work, right?
Second, wouldn't it make sense to do `SupplyManager<Supply_>` directly?

Reply:
1) Yes, was a mistake
2) No: SupplyManager handles multiple Supple types.